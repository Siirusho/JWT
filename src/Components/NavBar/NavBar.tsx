import { alpha, AppBar, Avatar, Badge, InputBase, makeStyles, Toolbar, Typography, Theme, createStyles } from "@material-ui/core";
import { Mail, Notifications, Search } from "@material-ui/icons";
import { useState } from "react";
import { FC } from 'react';

const useStyles = makeStyles<Theme>(theme =>
    createStyles({
        searchButton: {
            display: 'none',
            [theme.breakpoints.down("sm")]: {
                display: 'inline-block',
                marginRight: theme.spacing(4),
            }
        },
        logoLg: {
            display: "none",
            [theme.breakpoints.up("sm")]: {
                display: "block",
            }
        },
        logoSm: {
            display: "block",
            [theme.breakpoints.up("sm")]: {
                display: "none"
            }
        },
        toolbar: {
            display: "flex",
            justifyContent: "space-between",
        },
        search: {
            display: "flex",
            alignItems: "center",
            backgroundColor: alpha(theme.palette.common.white, 0.15),
            '&:hover': {
                backgroundColor: alpha(theme.palette.common.white, 0.25),
            },
            borderRadius: theme.shape.borderRadius,
            transition: "0.6s ease",
            [theme.breakpoints.down("sm")]: {
                display: "none",
            }
        },
        searchIcon: {
            marginLeft: theme.spacing(1)
        },
        active: {
            width: "50%",
        },
        disable: {
            width: "25%",
        },
        input: {
            color: "white",
            marginLeft: theme.spacing(2)
        },
        badgeContainer: {
            display: "flex",
            alignItems: "center",
        },
        badge: {
            marginRight: theme.spacing(2),
        },
        navBar: {
            position: "fixed",
            top: 0,
            width: "100%",
        }
    }));

const NavBar: FC = () => {
    const classes = useStyles();
    const [toggle, setToggle] = useState(false);
    return (
        <>
            <AppBar className={classes.navBar}>
                <Toolbar className={classes.toolbar}>
                    <Typography variant="h6" className={classes.logoLg} style={{ cursor: "pointer" }}>
                        Home Page
                    </Typography>
                    <Typography variant="h6" className={classes.logoSm}>
                        Home
                    </Typography>
                    <div onClick={() => setToggle(!toggle)} className={toggle ? `${classes.active} ${classes.search}` : `${classes.disable} ${classes.search}`}>
                        <Search className={classes.searchIcon} />
                        <InputBase
                            placeholder="Search..."
                            className={classes.input}
                        />
                    </div>
                    <div className={classes.badgeContainer}>
                        <Search className={classes.searchButton} />
                        <Badge badgeContent={4} overlap={"rectangular"} color="secondary" className={classes.badge}>
                            <Mail />
                        </Badge>
                        <Badge badgeContent={2} overlap={"rectangular"} color="secondary" className={classes.badge}>
                            <Notifications />
                        </Badge>
                        <Avatar alt="Page Logo" src="" />
                    </div>
                </Toolbar>
            </AppBar>
        </>
    );
}

export default NavBar;
