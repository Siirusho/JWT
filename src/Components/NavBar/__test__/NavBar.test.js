import React from "react";
import NavBar from "../NavBar";
import { render } from "@testing-library/react";


describe("NavBar", () => {
    it("should render correct NavBar component and match with snapshot", () => {
        const { asFragment } = render(<NavBar />);
        expect(asFragment(<NavBar />)).toMatchSnapshot();
    })
});