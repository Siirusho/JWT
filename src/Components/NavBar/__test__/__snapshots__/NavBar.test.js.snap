// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render correct NavBar component and match with snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-navBar-12 mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-4 MuiToolbar-gutters"
    >
      <h6
        class="MuiTypography-root makeStyles-logoLg-2 MuiTypography-h6"
        style="cursor: pointer;"
      >
        Home Page
      </h6>
      <h6
        class="MuiTypography-root makeStyles-logoSm-3 MuiTypography-h6"
      >
        Home
      </h6>
      <div
        class="makeStyles-disable-8 makeStyles-search-5"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-searchIcon-6"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
        <div
          class="MuiInputBase-root makeStyles-input-9"
        >
          <input
            class="MuiInputBase-input"
            placeholder="Search..."
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="makeStyles-badgeContainer-10"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-searchButton-1"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
        <span
          class="MuiBadge-root makeStyles-badge-11"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
          <span
            class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangular MuiBadge-colorSecondary"
          >
            4
          </span>
        </span>
        <span
          class="MuiBadge-root makeStyles-badge-11"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
            />
          </svg>
          <span
            class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangular MuiBadge-colorSecondary"
          >
            2
          </span>
        </span>
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiAvatar-fallback"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
